### ISSUE TITLE
Terrible performance on Fire Tablets

### ISSUE THREAD

**Issue by madasus** [thomasloven/lovelace-auto-entities#379]
Hello,

I've been trying to figure out why my fire tablets are running slowly (i know - they are always slow) but i believe it is being caused by the auto-entities card.  When I remove it from my dashboard the performance returns to normal.

I am running v1.12.1

Here is my YAML.  Any ideas why this might be causing slowness? I have a lot of entities in my instance and the goal of this it to show me open security related items (doors, windows, garage doors, etc)

Is there a good way to optimize?

Thank you

M

```
type: custom:auto-entities
card:
  type: entities
  title: House Security
filter:
  include:
    - device: omnilink
      domain: binary_sensor
      state: 'on'
    - domain: cover
      state: open
    - domain: lock
      state: unlocked
    - domain: lock
      state: locking
  exclude:
    - domain: binary_sensor
      attributes:
        device_class: motion
    - entity_id: binary_sensor.light_sensor
    - entity_id: binary_sensor.generator_on
    - entity_id: cover.*somfy_shade*
sort:
  method: last_updated

```

**Comment by ildar170975** [comment]
Test with `last-changed`.

**Comment by ildar170975** [comment]
What about a normal desktop PC? Does it have slow performance too?
Also, test w/o "device" entities.

**Comment by madasus** [comment]
@ildar170975 The performance issues are not noticeable on my desktop.

I removed the device rows  (the three below) and the performance does appear much better.  Is this expected?

    - device: omnilink
      domain: binary_sensor
      state: 'on'

**Comment by madasus** [comment]
I think your on to something.

I switched to try sensor instead of binary_sensor (as my alarm panel has both) and the performance issue was still there when using the device filter.

Then I switched to this

type: custom:auto-entities
card:
  type: entities
  title: House Security
filter:
  include:
    - domain: sensor
      state: Not Ready
    - domain: cover
      state: open
    - domain: lock
      state: unlocked
    - domain: lock
      state: locking
  exclude:
    - domain: binary_sensor
      attributes:
        device_class: motion
    - entity_id: binary_sensor.light_sensor
    - entity_id: sensor.generator_on
    - entity_id: cover.*somfy_shade*
sort:
  method: last_changed

Basically removed the device filter and just said to include the sensor domain and the performance is really good again (even though I have extra items being returned that I don't want).  Is there just a bug with how the device filter is applying? or is this a known performance issue?

**Comment by ildar170975** [comment]
1. The code you posted is unformatted. Suggest to edit it.
2. Rename the issue into smth like "Using a `device` filter slows down a performance" if you think that this is a reason.

**Comment by madasus** [comment]
Thanks - updated both.

**Comment by mash2k3** [comment]
Yes this is correct, been having same performance issues. I was able to improve it by minimizing how many entities i have on the exclude filter. I was able to accomplish this by using - hidden_by: '*' filter and disabling visibility on entities i dont want showing up in auto-entities. It was especially slow when editing auto-entities cards now it is much better.
